### ISSUE TITLE
Feature Request: add 'assert_any_call', 'assert_called_once', ... helper methods

### ISSUE THREAD

**Issue by bcollazo** [getsentry/responses#236]
Right now, to assert calls were made as expected one must explicitly state the _index_ of the call.
```
assert responses.calls[0].request.url == 'http://twitter.com/api/1/foobar'
assert responses.calls[0].response.text == '{"error": "not found"}'
```

This can make for brittle tests, since sometimes the _order_ in which calls are made doesn't really matter. 

It'd be nice if the library provided methods like mock's [assert_any_call](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_any_call), [assert_called_once](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once), ... so that one can write tests that do not depend in the _order_ in which calls are made (implementation details).

**Comment by sanketplus** [comment]
@markstory would you be open to review a PR on this?

**Comment by sanketplus** [comment]
this should be closed by #318
